File: src/test/java/com/projecttemplategroup/projecttemplateartifact/security/AccessLogicAsAdmin.java
package com.projecttemplategroup.projecttemplateartifact.security;

import com.projecttemplategroup.projecttemplateartifact.BaseSpringTest;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

class AccessLogicAsAdmin extends BaseSpringTest {

    @Test
    void success_ping_public_api_as_admin() {
        ResponseEntity<String> resp = restTemplate.exchange("/v1/ping", HttpMethod.GET, new HttpEntity<String>(cookieHeader("test-token-admin")), String.class);

        assertEquals(HttpStatus.OK, resp.getStatusCode());
        assertEquals("public api pong", resp.getBody());
    }

    @Test
    void success_ping_admin_api_as_admin() {
        ResponseEntity<String> resp = restTemplate.exchange("/v1-adm/ping", HttpMethod.GET, new HttpEntity<String>(cookieHeader("test-token-admin")), String.class);

        assertEquals(HttpStatus.OK, resp.getStatusCode());
        assertEquals("admin pong", resp.getBody());
    }

    @Test
    void fail_ping_admin_api_with_invalid_token() {
        ResponseEntity<Map> resp = restTemplate.exchange("/v1-adm/ping", HttpMethod.GET, new HttpEntity<String>(cookieHeader("aaa")), Map.class);

        assertEquals(HttpStatus.UNAUTHORIZED, resp.getStatusCode());
        assertEquals("", resp.getBody().get("message"));
    }

    @Test
    void fail_ping_user_api_as_admin() {
        ResponseEntity<Map> resp = restTemplate.exchange("/v1-usr/ping", HttpMethod.GET, new HttpEntity<String>(cookieHeader("test-token-admin")), Map.class);

        assertEquals(HttpStatus.FORBIDDEN, resp.getStatusCode());
        assertEquals("", resp.getBody().get("message"));
    }


}

File: src/test/scala/com/projecttemplategroup/projecttemplateartifact/security/AccessLogicAsAdmin.scala
package com.projecttemplategroup.projecttemplateartifact.security

import com.projecttemplategroup.projecttemplateartifact.BaseSpringTest
import org.springframework.http.{HttpEntity, HttpMethod, HttpStatus, ResponseEntity}

class AccessLogicAsAdmin extends BaseSpringTest {

  val adminToken = "test-token-admin"

  Feature("check admin cookie token") {

    Scenario("success call public api as admin") {
      val resp = restTemplate.exchange("/v1/ping", HttpMethod.GET, new HttpEntity[String](cookieHeader(adminToken)), classOf[String])

      resp.getStatusCode shouldBe HttpStatus.OK
      resp.getBody shouldBe "public api pong"
    }

    Scenario("success call admin api as admin") {
      val resp: ResponseEntity[String] = restTemplate.exchange("/v1-adm/ping", HttpMethod.GET, new HttpEntity[String](cookieHeader(adminToken)), classOf[String])

      resp.getStatusCode shouldBe HttpStatus.OK
      resp.getBody shouldBe "admin pong"
    }

    Scenario("fail to call admin api with invalid token") {
      val resp = restTemplate.exchange("/v1-adm/ping", HttpMethod.GET, new HttpEntity[String](cookieHeader("aaa")), classOf[String])

      resp.getStatusCode shouldBe HttpStatus.UNAUTHORIZED
      jsonAssert(
        """{
          |"status":401,
          |"error":"Unauthorized",
          |"message":""}""".stripMargin, resp.getBody, false)
    }

    Scenario("fail to call user api as admin") {
      val resp = restTemplate.exchange("/v1-usr/ping", HttpMethod.GET, new HttpEntity[String](cookieHeader(adminToken)), classOf[String])

      resp.getStatusCode shouldBe HttpStatus.FORBIDDEN
      jsonAssert(
        """{
          |"status":403,
          |"error":"Forbidden",
          |"message":""}""".stripMargin, resp.getBody, false)
    }

  }
}
